src/util/string_stream.h

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) 2020 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_UTIL_STRING_STREAM_H
+#define BITCOIN_UTIL_STRING_STREAM_H
+
+#include <string>
+#include <sstream>
+#include <type_traits>
+#include <locale>
+
+/**
+ * Locale-independent version of std::to_string
+ */
+template <typename T>
+typename std::enable_if<std::is_integral<T>::value, std::string>::type ToString(const T t) {
```

Comment:
Remove the `enable_if`: we need this for all types (the floating points in particular).

That `ToString` doesn't print a lot of trailing zeroes is a feature and not a bug. You can note the difference in the comment though :)

Since `std::to_string` is locale dependent there should be no assumption regarding formatting regardless, so no harm done by not printing trailing zeroes. Also all current code sites are safe.